addons/binding/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/handler/SpotifyBridgeHandler.java

```diff
@@ -0,0 +1,605 @@
+                    scheduler.schedule(this::scheduledPollingRestart, POLL_DELAY_AFTER_COMMAND_S, TimeUnit.SECONDS);
+                }
+            } catch (SpotifyException e) {
+                logger.debug("Handle Spotify command failed: ", e);
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
+            }
+        }
+    }
+
+    @Override
+    public void dispose() {
+        active = false;
+        if (oAuthService != null) {
+            oAuthService.removeAccessTokenRefreshListener(this);
+        }
+        oAuthFactory.ungetOAuthService(thing.getUID().getAsString());
+        cancelSchedulers();
+    }
+
+    @Override
+    public ThingUID getUID() {
+        return thing.getUID();
+    }
+
+    @Override
+    public String getLabel() {
+        return thing.getLabel() == null ? "" : thing.getLabel().toString();
+    }
+
+    @Override
+    public boolean isAuthorized() {
+        try {
+            return oAuthService != null && oAuthService.getAccessTokenResponse() != null
+                    && oAuthService.getAccessTokenResponse().getAccessToken() != null
+                    && oAuthService.getAccessTokenResponse().getRefreshToken() != null;
+        } catch (OAuthException | IOException | OAuthResponseException | RuntimeException e) {
+            return false;
+        }
+    }
+
+    @Override
+    public String getUser() {
+        return thing.getProperties().get(PROPERTY_SPOTIFY_USER);
+    }
+
+    @Override
+    public boolean isOnline() {
+        return thing.getStatus() == ThingStatus.ONLINE;
+    }
+
+    @Nullable
+    SpotifyApi getSpotifyApi() {
+        return spotifyApi;
+    }
+
+    @Override
+    public boolean equalsThingUID(String thingUID) {
+        return getThing().getUID().getAsString().equals(thingUID);
+    }
+
+    @Override
+    public String formatAuthorizationUrl(String redirectUri) {
+        try {
+            return oAuthService.getAuthorizationUrl(redirectUri, null, thing.getUID().getAsString());
+        } catch (OAuthException e) {
+            logger.debug("Error constructing AuthorizationUrl: ", e);
+            return "";
+        }
+    }
+
+    @Override
+    public String authorize(String redirectUri, String reqCode) {
+        try {
+            logger.debug("Make call to Spotify to get access token.");
+            final AccessTokenResponse credentials = oAuthService.getAccessTokenResponseByAuthorizationCode(reqCode,
+                    redirectUri);
+            final String user = updateProperties(credentials);
+            logger.debug("Authorized for user: {}", user);
+            startPolling();
+            return user;
+        } catch (RuntimeException | OAuthException | IOException e) {
+            logger.debug("Authorize failed: ", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, e.getMessage());
+            throw new SpotifyException(e.getMessage());
```

Comment:
Should you update the status, log and rethrow? Please apply passing the cause here and not logging when rethrowing.

Reply:
I've removed the logging here. It still should update the status. This method is called from the servlet to pass authorization. If that failed this is reflected in the status so the user knows what is wrong. Because the error message itself is only known here the status must be updated here.